agents/src/worker.ts

```diff
@@ -331,6 +331,11 @@ export class Worker {
           this.#runWS(this.#session);
           return;
         } catch (e) {
+          if (typeof e !== 'string') {
+            // it's probably an Error type
```

Comment:
instead of assuming this another type checking for error instanceof Error would be a lot cleaner